client/request.go

```diff
@@ -0,0 +1,93 @@
+package client
+
+import (
+	log "github.com/cihub/seelog"
+	"github.com/nu7hatch/gouuid"
+
+	"github.com/b2aio/typhon/errors"
+)
+
+type Request interface {
```

Comment:
This is mainly because I'm planning on removing the `response` concrete type from the `client` package, and pushing it into the underlying transport where we unmarshal from the delivery into a `Response` - allowing us to return a `Response` interface. Having only one of `Request` and `Response` as an interface made the `Req` method interfaces suck.